src/modules/parser/parser.js

```diff
@@ -0,0 +1,184 @@
+                const itemStats = itemsStats[itemname];
+                if (
+                    !this.itemsStats[itemname] ||
+                    itemStats.isFile() &&
+                    this.itemsStats[itemname] &&
+                    itemStats.mtimeMs > this.itemsStats[itemname].mtimeMs
+                ) {
+                    const setFileChanges = async () => {
+                        const parsedFileData = await this.getParsedData(itemname);
+                        if (parsedFileData) {
+                            changes[itemname] = parsedFileData;
+                        }
+                    }
+                    changesPromises.push(setFileChanges());
+                }
+            }
+            Promise.all(changesPromises)
+                .then(() => res(changes))
+                .catch((err) => rej(err));
+        });
+    }
+
+    getParsedData(filename) {
+        return new Promise(async (res, rej) => {
+            const filepath = path.resolve(this.dirpath, filename);
+            const fileExtName = path.extname(filepath);
+            const fileData = await this.readFile(filepath);
+            let parsedData;
+            switch (fileExtName) {
+                case '.csv':
+                    parsedData = await this.parseCsv(fileData);
+                    break;
+                case '.xml':
+                    parsedData = await this.parseXml(fileData);
+                    break;
+            }
+            res(parsedData);
+        });
+    }
+
+    parseCsv(fileData) {
+        return new Promise(async (res, rej) => {
+            if (this.sync) {
+                try {
+                    const parsedData = parseCsvSync(fileData);
+                    res(parsedData);
+                } catch (err) {
+                    rej(err);
+                }
+            } else {
+                parseCsv(fileData, (err, parsedData) => err ? rej(err) : res(parsedData));
+            }
+        });
+    }
+
+    parseXml(fileData) {
+        return new Promise((res, rej) => {
+            try {
+                const parsedData = xmlParser.parse(fileData.toString());
+                res(parsedData);
+            } catch (err) {
+                rej(err);
+            }
+        });
+    }
+
+    getStats(itempath) {
+        return new Promise((res, rej) => {
+            if (this.sync) {
+                try {
+                    const stats = fs.statSync(itempath);
+                    res(stats);
+                } catch (err) {
+                    rej(err);
+                }
+            } else {
+                fs.stat(itempath, (err, stats) => err ? rej(err) : res(stats));
+            }
+        });
+    }
+
+    readFile(filepath) {
+        return new Promise((res, rej) => {
+            if (this.sync) {
+                fs.readFile(filepath, (err, data) => err ? rej(err) : res(data));
+            } else {
+                const data = fs.readFileSync(filepath);
```

Comment:
Here you still return a promise which will be moved out of the execution stack and handled in the event queue which is not truly 'sync' execution.